controllers/machinehealthcheck_targets.go

```diff
@@ -120,6 +120,12 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 
 	// the node has not been set yet
 	if t.Node == nil {
+		if isZero(timeoutForMachineToHaveNode) {
```

Comment:
WDYT does it make sense to check if it's equal to `disabledNodeStartupTimeout`? I know it's essentially the same, but the dependency is a bit implicit here.

Reply:
Yeah I like that, makes it more explicit. I've updated :)